app/uk/gov/hmrc/eventhub/service/PublisherService.scala

```diff
@@ -75,7 +75,7 @@ class PublisherService @Inject() (
     mongoSetup.topics.find(_.name == topic) match {
       case None                                     => Left(NoEventTopic("No such topic"))
       case Some(topic) if topic.subscribers.isEmpty => Left(NoSubscribersForTopic("No subscribers for topic"))
-      case Some(topic) if !hasMandatoryPath(event, topic.subscribers) =>
+      case Some(topic) if !matchingPath(event, topic.subscribers) =>
         Left(NoMandatoryPath("Payload is missing mandatory path defined in config"))
```

Comment:
also rename the case class :)

Reply:
The error message should perhaps read `The payload does not match any path defined in config`